iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -328,4 +334,25 @@ public static PartitionData toPartitionData(StructLike key, Types.StructType key
     }
     return data;
   }
+
+  public static void getFiles(Table table, List<DataFile> dataFiles, List<DeleteFile> deleteFiles) {
+    CloseableIterable<FileScanTask> fileScanTasks = table.newScan().planFiles();
+    Set<String> dataFilesPath = Sets.newHashSet();
+    Set<String> deleteFilesPath = Sets.newHashSet();
+
+    for (FileScanTask fileScanTask : fileScanTasks) {
+      // filter repeated data files
```

Comment:
Under what cases do we have repeated files?

Reply:
I am not sure under which cases there can be repeated files, I added that because I saw such duplicates prevention when getting data and delete files in a code of another engine which also implements Iceberg compaction.